### DISCUSSION TITLE
HPC module loading

### DISCUSSION THREAD

**Discussion by edg-13** [firedrakeproject/firedrake#2994] [Firedrake support]
Hello,
I'm trying to run a Firedrake simulation using the Imperial HPC facility but I am unsure how to load the required modules in the job script, or create a personal conda environment with Firedrake installed (since I am not sure whether I have permission to manually install Firedrake).

Any help is appreciated.

**Comment by connorjward** [comment]
Please check out our [wiki entry on installing to HPC](https://github.com/firedrakeproject/firedrake/wiki/Running-on-HPC). If you still have problems @JDBetteridge should be able to advise.

Also note that Firedrake is [not compatible with Anaconda](https://www.firedrakeproject.org/download.html#system-anti-requirements) at present.

**Reply by JDBetteridge** [reply]
I'm not familiar with Imperial's HPC, but the guide Connor mentions is a good start. If singularity is supported, the singularity images are a nice zero-installation path to getting started.

**Reply by edg-13** [reply]
It looks like it might be, where can I find the singularity images?

**Reply by JDBetteridge** [reply]
You need to convert the docker images following this guide here (steps 1 & 2): https://github.com/firedrakeproject/firedrake/wiki/singularity

**Reply by edg-13** [reply]
Thank you, would the example job script be similar, e.g same global variables?

On another note, I have noticed firedrake is a loadable module on the HPC system. Using just `module load firedrake` in the job script gives me this error.
![image](https://github.com/firedrakeproject/firedrake/assets/86772327/26a090ec-6091-4299-9349-a43944f3df84)
If there's an easy fix for this, then I think this may have a cleaner job script.
But if using the singularity image is less fiddly, I'm happy to go ahead with that.

**Reply by JDBetteridge** [reply]
That screenshot looks like a Firedrake install from 2016, it almost certainly won't work.

The environment variables (=? global variables) all need to change to the ones on the HPC system, without access I can't tell you what these are, you will have to check the HPC documentation.

**Reply by edg-13** [reply]
I think in that case the spack method may be the easiest. I am running into this error when running `spack install`
![image](https://github.com/firedrakeproject/firedrake/assets/86772327/58569015-eaee-40eb-a22e-a683ef3e7346)

What am I missing? Here are my .spack/packages.yaml and <env>/spack.yaml files as txt files for github.
[spack.txt](https://github.com/firedrakeproject/firedrake/files/11798809/spack.txt)
[packages.txt](https://github.com/firedrakeproject/firedrake/files/11798811/packages.txt)

**Reply by JDBetteridge** [reply]
The Spack recipe is in serious need of work and has not been updated for a long time. Fixing the Spack installation is on my very long to do list
